tests/modules/ecs6-class/line.test.js

```diff
@@ -0,0 +1,118 @@
+const Point = require("../../../modules/ecs6-class/point");
+const Line = require("../../../modules/ecs6-class/line");
+
+describe('Line', () => {
+    const mockConstructorLine = jest.fn(constructor);
+    const mockConstructorPoint = jest.fn(constructor);
+
+    const line = new Line({ point1: new Point({ x: 1, y: 1 }), point2: new Point({ x: 2, y: 2 }) });
+    line.calculateSlope();
+    line.calculateNOfLineFunction();
+
+    describe('Line Constructor Tests', () => {
+        it('should create a Line with default points if none provided', () => {
+            const line = new Line({});
+            expect(line.point1).toBeInstanceOf(Point);
+            expect(line.point2).toBeInstanceOf(Point);
+            expect(line.slope).toBeUndefined()
+            expect(line.n).toBeUndefined()
+        });
+        it('should create a Line with specified points', () => {
+            const point1 = new Point({ x: 1, y: 2 });
+            const point2 = new Point({ x: 3, y: 4 });
+            const line = new Line({ point1, point2, slope: 2, n: 1 });
+            expect(line.point1).toBe(point1);
+            expect(line.point2).toBe(point2);
+            expect(line.slope).toBe(2)
+            expect(line.n).toBe(1)
+        });
+        it('should create a Line with one point and default the other properties', () => {
+            const point = new Point({ x: 1, y: 2 });
+            const line = new Line({ point1: point });
+            expect(line.point1).toBe(point);
+            expect(line.point2).toBeInstanceOf(Point);
```

Comment:
and what are the value of point2?

Reply:
I didn't understand what the problem is, it accepts the default values ​​of Point

Reply:
You are correct about this